aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1327 @@
+        simple_map::add(&mut metadata, utf8(IS_MULTI_STEP_PROPOSAL_KEY), to_bytes(&is_multi_step_proposal));
+
+        let is_multi_step_in_execution_key = utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        if (is_multi_step_proposal) {
+            // If the given proposal is a multi-step proposal, we will add a flag to indicate if this multi-step proposal is in execution.
+            // This value is by default false. We turn this value to true when we start executing the multi-step proposal. This value
+            // will be used to disable further voting after we started executing the multi-step proposal.
+            simple_map::add(&mut metadata, is_multi_step_in_execution_key, to_bytes(&false));
+            // If the proposal is a single-step proposal, we check if the metadata passed by the client has the IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY key.
+            // If they have the key, we will remove it, because a single-step proposal that doesn't need this key.
+        } else if (simple_map::contains_key(&mut metadata, &is_multi_step_in_execution_key)) {
+            simple_map::remove(&mut metadata, &is_multi_step_in_execution_key);
+        };
+
+        table::add(&mut voting_forum.proposals, proposal_id, Proposal {
+            proposer,
+            creation_time_secs: timestamp::now_seconds(),
+            execution_content: option::some<ProposalType>(execution_content),
+            execution_hash,
+            metadata,
+            min_vote_threshold,
+            voters,
+            expiration_secs,
+            yes_votes: 0,
+            no_votes: 0,
+            voted_records: table::new(),
+            is_resolved: false,
+            resolution_time_secs: 0,
+        });
+
+        if (std::features::module_event_migration_enabled()) {
+            event::emit(
+                CreateProposal {
+                    proposal_id,
+                    execution_hash,
+                    expiration_secs,
+                    metadata,
+                    min_vote_threshold,
+                },
+            );
+        };
+        event::emit_event<CreateProposalEvent>(
+            &mut voting_forum.events.create_proposal_events,
+            CreateProposalEvent {
+                proposal_id,
+                execution_hash,
+                expiration_secs,
+                metadata,
+                min_vote_threshold,
+            },
+        );
+
+        proposal_id
+    }
+
+    /// Vote on the given proposal.
+    ///
+    /// @param _proof Required so only the governance module that defines ProposalType can initiate voting.
+    /// @param voter signer of the voter.
+    /// @param voting_forum_address The address of the forum where the proposals are stored.
+    /// @param proposal_id The proposal id.
+    /// @param should_pass Whether the votes are for yes or no.
+    public fun vote<ProposalType: store>(
+        _proof: &ProposalType,
+        voter: &signer,
```

Comment:
Is `_proof : &ProposalType` also the first argument in `voting` module? Typically `signer` is the first argument. Please check.

Reply:
Since this `vote` function is not an entry function, the order of the signer parameter does not matter. The sequence depends on how we call it, and we need to maintain the same order.

However, it would be better if the signer parameter is in the first place